zerver/lib/subscription_info.py

```diff
@@ -438,14 +438,20 @@ def bulk_get_subscriber_user_ids(
     is_subscribed: bool
     check_user_subscribed = lambda user_profile: is_subscribed
 
+    has_atleast_one_private_channel = False
+    for stream_dict in stream_dicts:
+        if stream_dict["invite_only"]:
+            has_atleast_one_private_channel = True
+            break
```

Comment:
I think this isn't the right check; if the acting user is a guest, we need to also consider non-web-public channels. 

Thinking about this a bit more, I think the right algorithm might be to, rather than passing `user_recursive_group_ids` into `validate_user_access_to_subscribers_helper`, pass a simple `dataclass` called something like `UserMembershipDetails` containing just a `user_recursive_group_ids` field with the `set[int] | None` type, and if `validate_user_access_to_subscribers_helper` needs to fetch it, it does so and stores it onto the `dataclass` it was provided, meaning that future calls can naturally use the fact that this value has already been computed, without it being important to write a correct check for whether or not it might be used.